src/functions/utils.php

```diff
@@ -987,6 +987,54 @@ function tribe_get_request_vars( $refresh = false ) {
 	}
 }
 
+if ( ! function_exists( 'tribe_localize_script' ) ) {
+	/**
+	 * `wp_localize_script` just adds new data as a new line item regardless of whether or not the
+	 * object exists. This is not ideal, so we check and see if any localized data already exists
+	 * for our script. Then we edit existing elements rather than duplicating them.
+	 *
+	 * @since TBD
+	 *
+	 * @param string|object $enqueued_script The script handle we're saving data for - must already be loaded!
+	 * @param array         $localized_data The data we want to save.
+	 *
+	 * @return boolean Returns the result from wp_localize_script, or false on a failure.
+	 */
+	function tribe_localize_script( $enqueued_script, $localized_data = [] ) {
+		global $wp_scripts;
+
+		if ( ! $wp_scripts instanceof WP_Scripts) {
+			return false;
+		}
+
+		$data = $wp_scripts->get_data( $enqueued_script, 'data' );
+
+		if ( empty( $data ) ) {
+			// If no data, then just do the default.
+			return wp_localize_script( $enqueued_script, 'obj', $localized_data );
+		} else {
+			// If we have existing data, we try to append to it.
+			if ( ! is_array( $data ) ) {
+				$data = json_decode( str_replace( 'var obj = ', '', substr( $data, 0, -1 ) ), true );
+			}
+
+			// If it cannot be decoded or if the encoded data is deeper than
+			// the recursion limit, fall back to the default handling.
+			if ( empty( $data ) ) {
+				return wp_localize_script( $enqueued_script, 'obj', $localized_data );
+			}
+
+			foreach ( $data as $key => $value ) {
+				$localized_data[ $key ] = $value;
+			}
+
+			$wp_scripts->add_data( $enqueued_script, 'data', '' );
+
+			return wp_localize_script( $enqueued_script, 'obj', $localized_data );
```

Comment:
Whatever test you write, it will end up having to check on if, and with which values, the `wp_localize_script` function is called.

To test check what's in the `WP_Dependencies::registered` property (it's `public`) after the code executed.
Get hold of the global scripts object `wp_localize_script` will use by means of `$GLOBALS['wp_scripts']`.